src/Lykke.AzureStorage/Tables/Decorators/EncryptedTableStorageDecorator.cs

```diff
@@ -0,0 +1,461 @@
+        }
+
+        public Task DeleteAsync(IEnumerable<T> items)
+        {
+            var cryptoItems = items.Select(Encrypt);
+            return _storage.DeleteAsync(cryptoItems);
+        }
+
+        public Task<bool> CreateIfNotExistsAsync(T item)
+        {
+            var cryptoItem = Encrypt(item);
+            return _storage.CreateIfNotExistsAsync(cryptoItem);
+        }
+
+        public bool RecordExists(T item)
+        {
+            var cryptoItem = Encrypt(item);
+            return _storage.RecordExists(cryptoItem);
+        }
+
+        public Task<bool> RecordExistsAsync(T item)
+        {
+            var cryptoItem = Encrypt(item);
+            return _storage.RecordExistsAsync(cryptoItem);
+        }
+
+        public async Task<T> GetDataAsync(string partition, string row)
+        {
+            var cryptoItem = await _storage.GetDataAsync(partition, row);
+            return Decrypt(cryptoItem);
+        }
+
+        public async Task<IList<T>> GetDataAsync(Func<T, bool> filter = null)
+        {
+            var cryptoItems = await _storage.GetDataAsync(Map(filter));
+            return cryptoItems.Select(Decrypt).ToList();
+        }
+
+        public async Task<IEnumerable<T>> GetDataAsync(string partitionKey, IEnumerable<string> rowKeys, int pieceSize = 100, Func<T, bool> filter = null)
+        {
+            var cryptoItems = await _storage.GetDataAsync(partitionKey, rowKeys, pieceSize, Map(filter));
+            return cryptoItems.Select(Decrypt);
+        }
+
+        public async Task<IEnumerable<T>> GetDataAsync(IEnumerable<string> partitionKeys, int pieceSize = 100, Func<T, bool> filter = null)
+        {
+            var cryptoItems = await _storage.GetDataAsync(partitionKeys, pieceSize, Map(filter));
+            return cryptoItems.Select(Decrypt);
+        }
+
+        public async Task<IEnumerable<T>> GetDataAsync(IEnumerable<Tuple<string, string>> keys, int pieceSize = 100, Func<T, bool> filter = null)
+        {
+            var cryptoItems = await _storage.GetDataAsync(keys, pieceSize, Map(filter));
+            return cryptoItems.Select(Decrypt);
+        }
+
+        public Task GetDataByChunksAsync(Func<IEnumerable<T>, Task> chunks)
+        {
+            return _storage.GetDataByChunksAsync(Map(chunks));
+        }
+
+        public Task GetDataByChunksAsync(Action<IEnumerable<T>> chunks)
+        {
+            return _storage.GetDataByChunksAsync(Map(chunks));
+        }
+
+        public Task GetDataByChunksAsync(string partitionKey, Action<IEnumerable<T>> chunks)
+        {
+            return _storage.GetDataByChunksAsync(partitionKey, Map(chunks));
+        }
+
+        public Task<(IEnumerable<T> Entities, string ContinuationToken)> GetDataWithContinuationTokenAsync(TableQuery<T> rangeQuery, string continuationToken)
+        {
+            throw new NotSupportedException($"{nameof(EncryptedTableStorageDecorator<T>)} does not support Queries");
```

Comment:
Why?

Reply:
The problem is we cannot guarantee that this method will work for all queries. It will work for queries on non-encrypted fields only. Since each encrypted value has its own IV, we cannot map plain value to the encrypted value in DB.
We can pass the provided parameters to the same method in internal storage. It will work for some cases, but the result can be unexpected for called.